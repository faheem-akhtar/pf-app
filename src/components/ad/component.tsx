import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
