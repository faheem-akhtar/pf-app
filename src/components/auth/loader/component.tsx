import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
