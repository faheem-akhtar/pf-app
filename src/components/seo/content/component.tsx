import { FunctionComponent } from 'react';

import { domClassMerge } from 'helpers/dom/class-merge';

import { SeoContentComponentPropsInterface } from './component-props.interface';
import styles from './seo-content.module.scss';

export const SeoContentComponent: FunctionComponent<SeoContentComponentPropsInterface> = ({
  heading,
  content,
  image,
}) => (
  <div className={styles.container}>
    {heading && <h2 className={styles.heading}>{heading}</h2>}
    {content && (
      <div className={styles.body}>
        {image && (
          <picture
            className={domClassMerge(styles['body-picture'], {
              [styles['body-picture--right']]: !image.align || image.align === 'right',
              [styles['body-picture--left']]: image.align === 'left',
            })}
          >
            <img className={styles['body-picture-img']} src={image.url} alt={image.alt} />
          </picture>
        )}
        <div className={styles['body-html']} dangerouslySetInnerHTML={{ __html: content }} />
      </div>
    )}
  </div>
);
