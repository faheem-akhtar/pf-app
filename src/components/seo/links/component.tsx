import { FunctionComponent } from 'react';

import { AccordionComponent } from 'library/accordion/component';
import { AccordionItemComponent } from 'library/accordion/item/component';

import styles from '../seo.module.scss';
import { SeoLinksComponentPropsInterface } from './component-props.interface';

export const SeoLinksComponent: FunctionComponent<SeoLinksComponentPropsInterface> = ({
  popularSearches,
  nearbyAreas,
  alternateCategory,
}) => (
  <AccordionComponent>
    {popularSearches && (
      <AccordionItemComponent title={popularSearches.title} expanded>
        {popularSearches.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
        {alternateCategory?.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
      </AccordionItemComponent>
    )}

    {nearbyAreas && (
      <AccordionItemComponent title={nearbyAreas.title}>
        {nearbyAreas.links.map((link) => (
          <a key={link.path} href={link.path} className={styles.link}>
            {link.title}
          </a>
        ))}
      </AccordionItemComponent>
    )}
  </AccordionComponent>
);
