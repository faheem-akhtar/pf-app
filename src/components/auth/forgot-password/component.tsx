import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
