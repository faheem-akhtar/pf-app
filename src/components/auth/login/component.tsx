import React, { useState } from 'react';

import { AuthFacebookService } from 'services/auth/facebook.service';
import { AuthGoogleService } from 'services/auth/google.service';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthLoginPropsInterface } from 'components/auth/login/props.interface';
import { AuthLoginService } from 'services/auth/login.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { IconFacebookTemplate } from 'components/icon/facebook-template';
import { IconGoogleTemplate } from 'components/icon/google-template';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';
const captchaService = new GoogleRecaptchaService();

export const AuthLoginComponent = (props: AuthLoginPropsInterface): JSX.Element => {
  const { t } = useTranslation();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [AuthLoginFieldEnum.password]: [validationRequired(t('auth/empty-password'))],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  return (
    <div>
      <div className={styles.heading}>{t('sign-in')}</div>
      {/* Facebook */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.facebook)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthFacebookService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconFacebookTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-facebook')}
      </ButtonTemplate>
      {/* Google */}
      <ButtonTemplate
        type='button'
        className={domClassMerge(styles.button, styles.google)}
        componentType={ButtonComponentTypeEnum.tertiary}
        size={ButtonSizeEnum.small}
        onClick={(): void => {
          setIsLoading(true);
          AuthGoogleService.signIn()
            .then(props.onClose)
            .catch(() => {
              setErrorMessage(`${t('auth/something-wrong')}! ${t('auth/try-later')}`);
              props.onClose();
            })
            .finally(() => setIsLoading(false));
        }}
        icon={{ component: IconGoogleTemplate, position: ButtonIconPositionEnum.left }}
      >
        {t('auth/sign-in-google')}
      </ButtonTemplate>
      <div className={styles.divider}>
        <span className={styles.text}>{t('auth/or')}</span>
      </div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthLoginFieldEnum.email]: email,
              [AuthLoginFieldEnum.password]: password,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthLoginService({
                email,
                password,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                  } else {
                    setErrorMessage(e.error.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthLoginFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthLoginFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthLoginFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('password')}
            type='password'
            value={password}
            error={!!errors[AuthLoginFieldEnum.password]}
            onChange={(value): void => {
              !validate({
                [AuthLoginFieldEnum.password]: value,
              });
              setPassword(value);
            }}
            errorText={errors[AuthLoginFieldEnum.password]}
          />
        </div>
        <div className={styles['input-area']} onClick={props.onForgotPassword}>
          <div className={domClassMerge(styles.link, styles['forgot-password'])}>{t('auth/forgot-password')}?</div>
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('log-in')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
