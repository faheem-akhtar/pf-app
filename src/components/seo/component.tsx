import { FunctionComponent } from 'react';

import { SeoComponentPropsInterface } from './component-props.interface';
import { SeoContentComponent } from './content/component';
import { SeoLinksComponent } from './links/component';
import styles from './seo.module.scss';

export const SeoComponent: FunctionComponent<SeoComponentPropsInterface> = ({
  popularSearches,
  nearbyAreas,
  alternateCategory,
  content,
}) => (
  <section id='cms-content'>
    {(content?.primaryContent || content?.secondaryContent) && <hr className={styles.hr} />}
    {content?.primaryContent && (
      <SeoContentComponent
        heading={content.primaryHeading}
        content={content.primaryContent}
        image={
          content.primaryImageUrl
            ? {
                url: content.primaryImageUrl,
                alt: content.primaryImageAlt,
              }
            : undefined
        }
      />
    )}
    {content?.secondaryContent && (
      <SeoContentComponent
        heading={content.secondaryHeading}
        content={content.secondaryContent}
        image={
          content.secondaryImageUrl
            ? {
                url: content.secondaryImageUrl,
                alt: content.secondaryImageAlt,
                align: 'left',
              }
            : undefined
        }
      />
    )}
    <SeoLinksComponent
      popularSearches={popularSearches}
      nearbyAreas={nearbyAreas}
      alternateCategory={alternateCategory}
    />
  </section>
);
