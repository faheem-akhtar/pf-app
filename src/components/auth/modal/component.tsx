import React, { useState } from 'react';

import { AuthForgotPasswordComponent } from 'components/auth/forgot-password/component';
import { AuthLoginComponent } from 'components/auth/login/component';
import { AuthRegistrationComponent } from 'components/auth/registration/component';
import { AuthScreenEnum } from 'components/auth/screen.enum';
import { IconThickCrossTemplate } from 'components/icon/thick/cross-template';
import { functionNoop } from 'helpers/function/noop';

import styles from './auth-modal-component.module.scss';
import { AuthModalPropsInterface } from './props.interface';

export const AuthModalComponent = ({
  close = functionNoop,
  cancel = functionNoop,
  success = functionNoop,
}: AuthModalPropsInterface): JSX.Element => {
  const [authScreen, setAuthScreen] = useState(AuthScreenEnum.login);
  return (
    <div
      className={styles.content}
      onClick={(e): void => {
        e.stopPropagation();
      }}
    >
      <div className={styles.header}>
        <button
          onClick={(): void => {
            close();
            cancel();
          }}
        >
          <IconThickCrossTemplate class={styles.closeIcon} />
        </button>
      </div>
      {authScreen === AuthScreenEnum.login && (
        <AuthLoginComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
          onForgotPassword={(): void => setAuthScreen(AuthScreenEnum.forgotPassword)}
        />
      )}
      {authScreen === AuthScreenEnum.registration && (
        <AuthRegistrationComponent
          onClose={close}
          onSuccess={success}
          onLogin={(): void => setAuthScreen(AuthScreenEnum.login)}
        />
      )}
      {authScreen === AuthScreenEnum.forgotPassword && (
        <AuthForgotPasswordComponent
          onClose={close}
          onSuccess={success}
          onRegister={(): void => setAuthScreen(AuthScreenEnum.registration)}
        />
      )}
    </div>
  );
};
