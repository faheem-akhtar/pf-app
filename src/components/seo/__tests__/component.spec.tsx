import { render } from '@testing-library/react';

import { SeoComponent } from '../component';

describe('SeoComponent', () => {
  it('should render the component with primary content', () => {
    expect(
      render(
        <SeoComponent
          content={{
            primaryContent: '<p>This is param for primary content</p>',
            createdAt: '2021-01-01',
          }}
        />
      ).container
    ).toMatchSnapshot();
  });

  it('should render the component with primary content with image', () => {
    expect(
      render(
        <SeoComponent
          content={{
            primaryContent: '<p>This is param for primary content</p>',
            primaryImageUrl: 'https://www.propertyfinder.ae/dist/common/assets/favicon/favicon-32x32.png',
            createdAt: '2021-01-01',
          }}
        />
      ).container
    ).toMatchSnapshot();
  });

  it('should render the component with secondary content', () => {
    expect(
      render(
        <SeoComponent
          content={{
            secondaryContent: '<p>This is param for secondary content</p>',
            createdAt: '2021-01-01',
          }}
        />
      ).container
    ).toMatchSnapshot();
  });

  it('should render the component with secondary content with image', () => {
    expect(
      render(
        <SeoComponent
          content={{
            secondaryContent: '<p>This is param for secondary content</p>',
            secondaryImageUrl: 'https://www.propertyfinder.ae/dist/common/assets/favicon/favicon-32x32.png',
            createdAt: '2021-01-01',
          }}
        />
      ).container
    ).toMatchSnapshot();
  });
});
