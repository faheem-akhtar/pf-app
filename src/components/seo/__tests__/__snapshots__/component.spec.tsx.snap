// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeoComponent should render the component with primary content 1`] = `
<div>
  <section
    id="cms-content"
  >
    <hr
      class="hr"
    />
    <div
      class="container"
    >
      <div
        class="body"
      >
        <div
          class="body-html"
        >
          <p>
            This is param for primary content
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`SeoComponent should render the component with primary content with image 1`] = `
<div>
  <section
    id="cms-content"
  >
    <hr
      class="hr"
    />
    <div
      class="container"
    >
      <div
        class="body"
      >
        <picture
          class="body-picture body-picture--right"
        >
          <img
            class="body-picture-img"
            src="https://www.propertyfinder.ae/dist/common/assets/favicon/favicon-32x32.png"
          />
        </picture>
        <div
          class="body-html"
        >
          <p>
            This is param for primary content
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`SeoComponent should render the component with secondary content 1`] = `
<div>
  <section
    id="cms-content"
  >
    <hr
      class="hr"
    />
    <div
      class="container"
    >
      <div
        class="body"
      >
        <div
          class="body-html"
        >
          <p>
            This is param for secondary content
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`SeoComponent should render the component with secondary content with image 1`] = `
<div>
  <section
    id="cms-content"
  >
    <hr
      class="hr"
    />
    <div
      class="container"
    >
      <div
        class="body"
      >
        <picture
          class="body-picture body-picture--left"
        >
          <img
            class="body-picture-img"
            src="https://www.propertyfinder.ae/dist/common/assets/favicon/favicon-32x32.png"
          />
        </picture>
        <div
          class="body-html"
        >
          <p>
            This is param for secondary content
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
